File: app/src/main/java/com/mert/bilgram/SignInWithTabs.java
package com.mert.bilgram;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.PreferenceManager;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.app.AppCompatDelegate;
import androidx.viewpager.widget.ViewPager;

import com.google.android.material.tabs.TabLayout;
import com.google.firebase.auth.FirebaseAuth;

public class SignInWithTabs extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        SharedPreferences mPrefs = PreferenceManager.getDefaultSharedPreferences(this);
        if (mPrefs.getBoolean("NIGHT_MODE", false))
            AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES);
        else AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO);

        if (AppCompatDelegate.getDefaultNightMode() == AppCompatDelegate.MODE_NIGHT_YES)
            setTheme(R.style.AppThemeDark);

        if (FirebaseAuth.getInstance().getCurrentUser() != null)

            startActivity(new Intent(this, FeedActivity.class));

        setContentView(R.layout.activity_sign_in_with_tabs);
        ViewPager viewPager = findViewById(R.id.viewPager);
        TabLayout tabLayout = findViewById(R.id.tabLayout);
        TabAdapter adapter = new TabAdapter(getSupportFragmentManager());
        adapter.addFragment(new SignInFragment(), "SIGN IN");
        adapter.addFragment(new SignUpFragment(), "SIGN UP");
        viewPager.setAdapter(adapter);
        tabLayout.setupWithViewPager(viewPager);

    }

}

File: app/src/main/java/com/mert/bilgram/SignInWithTabs.kt
package com.mert.bilgram

import android.content.Intent
import android.content.res.Configuration
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.viewpager.widget.ViewPager
import com.google.android.material.tabs.TabLayout
import com.google.firebase.auth.FirebaseAuth

class SignInWithTabs : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        when (resources.configuration.uiMode and Configuration.UI_MODE_NIGHT_MASK) {
            Configuration.UI_MODE_NIGHT_YES -> setTheme(R.style.AppThemeDark)
            Configuration.UI_MODE_NIGHT_NO -> setTheme(R.style.AppThemeLight)
        }

        if (FirebaseAuth.getInstance().currentUser != null)
            startActivity(Intent(this, FeedActivity::class.java))

        setContentView(R.layout.activity_sign_in_with_tabs)

        val viewPager = findViewById<ViewPager>(R.id.viewPager)
        val adapter = TabAdapter(supportFragmentManager)

        adapter.addFragment(SignInFragment(), getString(R.string.sign_in))
        adapter.addFragment(SignUpFragment(), getString(R.string.sign_up))

        viewPager.adapter = adapter

        findViewById<TabLayout>(R.id.tabLayout).setupWithViewPager(viewPager)
    }
}